runtime/worker.rs

```diff
@@ -243,10 +246,33 @@ impl MainWorker {
 
   /// Create new inspector session. This function panics if Worker
   /// was not configured to create inspector.
-  pub fn create_inspector_session(&mut self) -> Box<InspectorSession> {
-    let inspector = self.inspector.as_mut().unwrap();
+  pub async fn create_inspector_session(&mut self) -> LocalInspectorSession {
+    let inspector = self.inspector.as_ref().unwrap();
+    let mut session = inspector.create_local_session();
+
+    {
+      let response =
+        session.post_message("Runtime.runIfWaitingForDebugger", None);
+      tokio::pin!(response);
+
+      loop {
+        tokio::select! {
+          result = &mut response => {
+            result.expect("Failed to start session");
+            break;
+          }
+
+          _ = self.run_event_loop() => {
+            // A zero delay is long enough to yield the thread in order to prevent the loop from
+            // running hot for messages that are taking longer to resolve like for example an
+            // evaluation of top level await.
+            tokio::time::sleep(tokio::time::Duration::from_millis(0)).await;
```

Comment:
This looks suspect. Why is it necessary?

Reply:
100% cpu usage otherwise.

Reply:
Because the "local session" is now run through `JsRuntimeInspector` you need to poll either the inspector or worker (which polls inspector internally). It's a bit unfortunate that I had to add it, but I managed to encapsulate it in a `Worker::with_event_loop` helper; I think this solution is acceptable if it lets us move inspector to `deno_core`.